src/application-base/ApplicationBase.jsx

```diff
@@ -106,19 +107,23 @@ const ApplicationBase = ({
         translationsLoadingPlaceholder={translationsLoadingPlaceholder}
         locale={locale}
       >
-        <ActiveBreakpointProvider>
-          <ApplicationErrorBoundary>
+        <ApplicationIntlProvider>
```

Comment:
Terra-base will render terra-i18n which will render the terra-i18n provider so we shoudn't need to define a new one

Reply:
The ApplicationIntlProvider here (and the ApplicationIntlContext.Provider that it renders) are different than the IntlProvider provided by terra-base. The ApplicationIntlProvider forwards the intl value from the IntlProvider through a modern Context. 

With Terra as a whole being stuck on react-intl v2.9 for the foreseeable future, I didn't care for the idea of consumers forced to interface with the legacy context APIs just for intl. With this in place, consumer's get the benefit of a cleaner API that's more consistent with all the other context-based APIs that terra-application introduces. Consumers of this should also feel no effects from the eventual bump to react-intl v3 (and the deprecation of `injectIntl`).

The new ApplicationIntlContext sits adjacent to the existing react-intl implementation. All the existing Terra components work the same as they do today. Consumers could opt to use react-intl directly, and it'd work just fine. But for a component that already has a dependency on terra-application, they'd get everything they need for i18n for free, which is nice.

I do understand that it's a divergence from the non-application Terra components today. I feel like the benefits outweigh the drawbacks in that regard, but I'm open to input here.

Reply:
Is this something we should be providing from terra-i18n instead of from terra-application?

Could this caused the situation were components developed and tested in terra-dev-site has access to intl via context, but then the deployed standalone components don't actually have access to it since they aren't wrapped by terra-application?

Reply:
I have thought about that. Doing so would introduce peer dependencies on terra-i18n for all packages that need translations, which wouldn't be passive in the short term. We could also include it directly within terra-base, but that would re-introduce dependencies on terra-base where we had them previously.

I think we could still make a decision to do so in the future, but even so, I'd expect terra-application to provide that new intl context, even if it's just a passthrough at that point. The API from terra-application can remain constant and passive for consumers of terra-application.

The alternative to what I have here would be to do nothing in the short term. I'd prefer to get folks on the right path and not set them up with technical debt from the get go.

Reply:
If terra-application is providing the context, are we assuming terra-base will be deprecated or only provided for base styles?

Reply:
> Could this caused the situation were components developed and tested in terra-dev-site has access to intl via context, but then the deployed standalone components don't actually have access to it since they aren't wrapped by terra-application?

Components that don't have a peerDependency on terra-application wouldn't use this context, which would be Terra's vast majority of components. They'd still be using react-intl directly (unless we decide to do something different in the future like mentioned above). We're still relying on there being a (single) react-intl instance in the bundle.

Reply:
Sorry these comments are out of order, I should refresh more often.

> If terra-application is providing the context, are we assuming terra-base will be deprecated or only provided for base styles?

I don't want to make any assumptions about terra-base. I think terra-base continues to have value as the barest-bones provider of Terra's global requirements (global styles and i18n) for all components in its ecosystem. 

Applications that want to only use Terra's non-framework components (say they really like Button or something) shouldn't be required to use terra-application's ApplicationBase and its associated complexities. In these cases, having terra-base available as a stripped down option seems valuable.

Reply:
Talked offline about this. This is the route we want to go! 👍

Reply:
:+1: I am going to revise the docs around the ApplicationIntlContext a bit to make consumption expectations more clear.